### PR TITLE
Db refactoring

### PR DESCRIPTION
**Pull request by lyle-adeptmind**

### CODE REVIEW

File: resources/cf/SUMMARY_IO_SPECS_CONFIG.json [hunk 1]
```diff
@@ -1,20 +1,21 @@
...
     "host": "192.168.0.109",
-    "port": "9200",
-    "index": "product-for-user-profile-job"
+    "port": "9200"
+  },
+  "indexes": {
+    "analytics": "adept-analytics-v2-app-*",
+    "products": "product-for-user-profile-job",
+    "products0": "cf-products-multi-mall-v3"
+  },
```
Comment (jinghe) [inline resources/cf/SUMMARY_IO_SPECS_CONFIG.json:15]: actually i still prefer to put everything about analytics data and everything about product data separated under their own umbrella, so we can potentially change the input data source (other than es index) easily

File: summary/compact.py [hunk 2]
```diff
@@ -99,28 +99,40 @@ def compact_docs(docs: List[Dict]):
     return [docs[0]]
 
 
-def compact(compact_collection, session_collection, output_collection):
-    import json
-    compact_kvs = [map_session_doc(doc) for doc in session_collection.find()]
+def compact(compact_collection, output_collection, daily_sessions):
```
Comment (jinghe) [inline summary/compact.py:102]: daily_sessions -> daily_session_collections?

File: resources/cf/SUMMARY_IO_SPECS_CONFIG.json (hunk 1 above)
Comment (lyle-adeptmind) [reply resources/cf/SUMMARY_IO_SPECS_CONFIG.json:15]: Except the idea is, let's say we have one database that has 2 indexes.  The idea is we want a repository for that database, and then that repository can be used to access either of those 2 indexes.  So then if we want to do that kind of thing, then we have to copy the database configuration in 2 places so the index and database config are together?  That kind of makes no sense.

Instead, we have the database configuration in one place, like "all product data on ES should go in this DB", and then the indexes should be separate, like "summary job uses this index, compact job uses that index", etc.

File: summary/compact.py (hunk 2 above)
Comment (lyle-adeptmind) [reply summary/compact.py:102]: This is changed in a different PR.

File: summary/compact.py [hunk 3]
```diff
@@ -99,28 +99,40 @@ def compact_docs(docs: List[Dict]):
...
+        for destination in [compacted_source, compacted_target]
+    ]
+    daily_source = [
         None if not destination else MongoClient()[destination["db"]][destination["table"]]
-        for destination in [compacted_source, daily_source, compacted_target]
+        for destination in daily_source
     ]
-    compact(compacted_source, daily_source, compacted_target)
+
+    compact(compacted_source, compacted_target, daily_source)
```
Comment (SiweiYang) [inline summary/compact.py:138]: would be great if you prompt with compacted_source, compacted_target, daily_source, and let user confirm before proceed

Comment (SiweiYang) [inline summary/compact.py:138]: would be great if you prompt with compacted_source, compacted_target, daily_source, and let user confirm before proceed

File: summary/config.py [hunk 4]
```diff
@@ -32,6 +32,8 @@
 )
 RAW_ANALYTICS_ES_SPEC = SUMMARY_IO_SPECS_CONFIG.get('analytics', {})
 PRODUCTS_ES_SPEC = SUMMARY_IO_SPECS_CONFIG.get('products', {})
+INDEXES_SPEC = SUMMARY_IO_SPECS_CONFIG.get('indexes', {})
```
Comment (SiweiYang) [inline summary/config.py:35]: not this PR, but do you think the naming can be improved?

File: summary/compact.py (hunk 3 above)
Comment (lyle-adeptmind) [reply summary/compact.py:138]: It's supposed to be run as a daily job on a cron schedule, so no?

File: summary/config.py (hunk 4 above)
Comment (lyle-adeptmind) [reply summary/config.py:35]: I don't mind adding it to this PR, what would you suggest for names?

Comment (SiweiYang) [reply summary/config.py:35]: i dont have an idea yet, just poking your thoughts. we can skip unless you have good names.

File: summary/compact.py (hunk 3 above)
Comment (SiweiYang) [reply summary/compact.py:138]: ok, if we do plan to support manual mode, can add that. with a flag -i maybe.(interactive mode)

Comment (lyle-adeptmind) [reply summary/compact.py:138]: Fair enough.  TBD for future.

File: summary/config.py (hunk 4 above)
Comment (lyle-adeptmind) [reply summary/config.py:35]: Nothing I can think of in particular.

**Pull request merged**